dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/net/info/CallbackNetworkInfoProvider.kt

```diff
@@ -104,13 +110,18 @@ internal class CallbackNetworkInfoProvider :
     }
 
     override fun getLatestNetworkInfo(): NetworkInfo {
-        return networkInfo
+        return lastNetworkInfo
     }
 
     // endregion
 
     // region Internal
 
+    private fun updateLastNetworkInfo(networkInfo: NetworkInfo) {
+        this.lastNetworkInfo = networkInfo
+        consentAwarePersister.persist(networkInfo)
+    }
```

Comment:
This could actually be done by overriding the setter of the property: 
```kotlin
private var lastNetworkInfo: NetworkInfo = NetworkInfo()
        set(value) {
            field = value
            consentAwarePersister.persist(value)
        }
```

This mean we don't have to remember to call this whenever we change the value of `lastNetworkInfo`